### PR TITLE
feat(play/URL-Shortner): add URL shortener component

### LINKED ISSUE #1259: [Add a Play]: URL Shortner
### What's the Play about?

✅ An URL shortener play using react

### What ReactJS concept will be used to create this play?

✅Components
✅Rest API
✅Axios
✅Custom Hook

### Record

- [X] I have checked the existing [issues](https://github.com/reactplay/react-play/issues)
- [X] I have read the [Contributing Guidelines](https://github.com/reactplay/react-play/blob/main/CONTRIBUTING.md)
- [X] I want to work on this issue

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by alaminrifat**
feat(play/feature-name): add URL shortener component

Added a new React component for URL shortening. This component allows users to enter a link to be shortened and displays the shortened link.

- Implemented the URL shortening functionality using the shrtco.de API.
- Added a user interface with an input field and button for shortening links.
- Display the shortened URL to the user.
- Included a "Copy URL to Clipboard" button for easy sharing.

Closes #1259

### CODE REVIEW

**Comment by alaminrifat** [comment]
@priyankarpal hello sir, I don't know what to do next, would you please me give an instruction?

**Comment by priyankarpal** [comment]
> @priyankarpal hello sir, I don't know what to do next, would you please me give an instruction?

we will review soon

**Comment by Abhishek-90** [comment]
@alaminrifat maybe we can remove the 2 console logs present, even though you are using the linter to ignore the consoles.

@priyankarpal let me know what you think?

**Comment by alaminrifat** [comment]
@Abhishek-90  Yes, the console log may be deprecated. I forgot to do this after completing all these.

File: src/plays/url-shortner/styles.css
Comment (priyankarpal) [inline src/plays/url-shortner/styles.css:1]: if you're not using this file please remove this file

**Review by priyankarpal — CHANGES_REQUESTED** [review]

File: src/plays/url-shortner/styles.css
Comment (alaminrifat) [reply src/plays/url-shortner/styles.css:1]: @priyankarpal style.css and console log has been removed ✅

Comment (priyankarpal) [reply src/plays/url-shortner/styles.css:1]: ok

File: package.json [hunk 1]
```diff
@@ -47,6 +47,7 @@
     "react-codemirror2": "^7.2.1",
     "react-color": "^2.19.3",
     "react-confetti": "^6.1.0",
+    "react-copy-to-clipboard": "^5.1.0",
```
Comment (Angryman18) [inline package.json:50]: remove this dependency. implement it with javascript. it need only fewer lines code only.

File: src/plays/url-shortner/UrlShortner.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (Angryman18) [inline src/plays/url-shortner/UrlShortner.js:1]: remove it.

File: src/plays/url-shortner/UrlShortner.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  // Your Code Start below.
+  const [userInput, setUserInput] = useState('');
+  const [shortenedLink, setShortenedLink] = useState('');
+
+  const handleSubmit = async () => {
+    try {
+      const response = await axios(`https://api.shrtco.de/v2/shorten?url=${userInput}`);
+      setShortenedLink(response.data.result.full_short_link);
+    } catch (e) {
+      // eslint-disable-next-line no-console
```
Comment (Angryman18) [inline src/plays/url-shortner/UrlShortner.js:17]: if the the userInput is not a valid url API gives error handle it in the catch block

File: src/plays/url-shortner/UrlShortner.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+      // console.log(e);
+    }
+  };
+
+  return (
+    <>
+      {/* <PlayHeader play={props} /> */}
```
Comment (Angryman18) [inline src/plays/url-shortner/UrlShortner.js:24]: One play must have this Playheader.

**Review by Angryman18 — CHANGES_REQUESTED** [review]
also remove all the eslint disable thing and all the console logs

File: package.json (hunk 1 above)
Comment (alaminrifat) [reply package.json]: This dependency has been removed.

File: src/plays/url-shortner/UrlShortner.js (hunk 2 above)
Comment (alaminrifat) [reply src/plays/url-shortner/UrlShortner.js]: Removed.

File: src/plays/url-shortner/UrlShortner.js (hunk 3 above)
Comment (alaminrifat) [reply src/plays/url-shortner/UrlShortner.js]: Error handling is done.

File: src/plays/url-shortner/UrlShortner.js (hunk 4 above)
Comment (alaminrifat) [reply src/plays/url-shortner/UrlShortner.js]: I have added this Playheader

File: package.json (hunk 1 above)
Comment (Angryman18) [reply package.json]: the onclick function to copy text could simply be `() => navigator.clipboard.writeText(shortenedLink);`

File: src/plays/url-shortner/UrlShortner.js [hunk 5]
```diff
@@ -1,59 +1,88 @@
...
-            <CopyToClipboard text={shortenedLink}>
-              <button className="mt-5 border-2 border-blue-500 text-blue-500 font-medium px-5 py-2 ml-4 rounded-md hover:bg-blue-500 hover:text-white">
+            <div className="mt-5">
+              {error && <div className="text-red-500 mb-2">{error}</div>}
+              <div className="relative px-4 py-2 bg-white shadow-lg rounded-xl  bg-clip-padding bg-opacity-60 border border-gray-200 text-slate-500">
+                {shortenedLink ? shortenedLink : 'Shorten An URL First'}
+              </div>
+              <button
+                className="mt-5 border-2 border-blue-500 text-blue-500 font-medium px-5 py-2 ml-4 rounded-md hover:bg-blue-500 hover:text-white"
+                onClick={() => copyTextToClipboard(shortenedLink)}
```
Comment (Angryman18) [inline src/plays/url-shortner/UrlShortner.js:81]: it could be this `onClick={() => navigator.clipboard.writeText(shortenedLink)}`

**Review by Angryman18 — APPROVED** [review]

File: src/plays/url-shortner/UrlShortner.js (hunk 5 above)
Comment (alaminrifat) [reply src/plays/url-shortner/UrlShortner.js:81]: I've tried this, In my case it wasn't working.

**Comment by alaminrifat** [comment]
@priyankarpal is there anything more I should do? please let me know.

**Review by priyankarpal — APPROVED** [review]

**Review by joshi-kaushal — APPROVED** [review]
LGTM

**Comment by atapas** [comment]
@alaminrifat vai I tried this using the preview link: https://deploy-preview-1261--reactplayio.netlify.app/plays/alaminrifat/url-shortner

It is giving me an error

image

Am I missing anything?

**Comment by rifat-incrementsinc** [comment]
@atapas  Sir, there may be an issue with the URL. I tried this URL with another free shortener, and it shows me this error 
image

**Comment by rifat-incrementsinc** [comment]
@atapas Sir, I've used a third party url shortener api and their server is down. I will wait 1/2 days if they don't recover I will use another. Thank you sir <3 
image

**Comment by atapas** [comment]
> @atapas Sir, I've used a third party url shortener api and their server is down. I will wait 1/2 days if they don't recover I will use another. Thank you sir <3 image

Sounds Good.

**Comment by priyankarpal** [comment]
@alaminrifat still has the same problem

**Comment by alaminrifat** [comment]
@priyankarpal I am fixing this

**Comment by alaminrifat** [comment]
@priyankarpal Would please kindly check now, I have fixed the issue. Thank you

**Comment by priyankarpal** [comment]
sure will check soon

**Review by priyankarpal — DISMISSED** [review]
Looks Good 🎉

**Review by atapas — CHANGES_REQUESTED** [review]
@alaminrifat The play header is not visible on this play. It could be due to the structural changes. ReactPlay automatically provides you a play header. You can check the code of other plays to see the required structure and fix accordingly.

image

**Comment by alaminrifat** [comment]
@atapas Sir, now it's ok 
image

**Review by joshi-kaushal — APPROVED** [review]

**Review by atapas — APPROVED** [review]
@alaminrifat Good Work!

**Comment by rifat-incrementsinc** [comment]
@atapas thank you sir

**Pull request merged**